server/src/main/java/com/octopus/teamcity/opentelemetry/server/helpers/OTELHelperFactory.java

```diff
@@ -0,0 +1,10 @@
+package com.octopus.teamcity.opentelemetry.server.helpers;
+
+import jetbrains.buildServer.serverSide.BuildPromotion;
+import jetbrains.buildServer.serverSide.SRunningBuild;
+
+public interface OTELHelperFactory {
```

Comment:
Added a factory, so we could externalise the caching of the OTELHelper, but also so we could pass a test specific one in from the tests